libs/ledger-live-common/src/api/socket.ts

```diff
@@ -91,45 +89,61 @@ export const createDeviceSocket = (
               nonce,
               apdu,
             });
-            // specific logic for detecting allow manager
-            let allowManagerAwaitingUser = false;
+
+            // Detect the specific exchange that triggers the allow secure channel request.
+            let pendingUserAllowSecureChannel = false;
 
             if (apdu.slice(0, 2).toString("hex") === "e051") {
-              timeoutForAllowManager = setTimeout(() => {
+              pendingUserAllowSecureChannel = true;
+              allowSecureChannelTimeout = setTimeout(() => {
                 if (unsubscribed) return;
-                allowManagerAwaitingUser = true;
                 o.next({
                   type: "device-permission-requested",
-                  wording: "Ledger Manager",
+                  wording: "Ledger Manager", // TODO make this dynamic per fw version to match device
                 });
-              }, ALLOW_MANAGER_DELAY);
+                // Nb Permission is only requested once per reboot, delaying the event
+                // prevents the UI from flashing the rendering for allowing.
+              }, ALLOW_SECURE_CHANNEL_DELAY);
             }
 
             const r = await transport.exchange(apdu);
 
-            if (timeoutForAllowManager) {
-              // if the exchange is faster than the timeout, we pass through because it's not "awaiting" user action
-              clearTimeout(timeoutForAllowManager as number);
-              timeoutForAllowManager = null;
+            if (allowSecureChannelTimeout) {
+              clearTimeout(allowSecureChannelTimeout as number);
+              allowSecureChannelTimeout = null;
             }
 
             if (unsubscribed) return;
             const status = r.readUInt16BE(r.length - 2);
 
-            // Pushes an error on locked device. TransportStatusError will
-            // be remapped to a LockedDeviceError with a correct error message.
-            if (status === StatusCodes.LOCKED_DEVICE) {
-              o.error(new TransportStatusError(status));
-              return;
-            }
+            let response;
+            switch (status) {
+              case StatusCodes.OK:
+                response = "success";
+                break;
 
-            // if allow manager was requested, we either throw if deny or emit accepted
-            if (allowManagerAwaitingUser) {
-              if (status === 0x6985 || status === 0x5501) {
-                o.error(new UserRefusedAllowManager());
+              case StatusCodes.LOCKED_DEVICE:
```

Comment:
I know the comment above was not amazing, because we don't just throw `LockedDeviceError`, but let's keep it:

```
// TransportStatusError will be remapped to a LockedDeviceError with a correct error message
```

Reply:
Sorry but the case literally tells you that without the comment, anyone reading that code should know or be able to look into what `TransportStatusError` is doing without the need for a comment.

<img width="352" alt="image" src="https://user-images.githubusercontent.com/4631227/209924330-1b5ae842-ca5b-46b9-9834-5d71b2efadd9.png">

Reply:
true !